dev/AppNotifications/WICUtility.cpp

```diff
@@ -0,0 +1,165 @@
+
+    if ((containerGuid == GUID_ContainerFormatBmp) && (bmpv == BMPV_5))
+    {
+        // Write the encoder option to the property bag instance.
+        VARIANT varValue{};
+
+        varValue.vt = VT_BOOL;
+        varValue.boolVal = VARIANT_TRUE;
+
+        // Options to enable the v5 header support for 32bppBGRA.
+        PROPBAG2 v5HeaderOption{};
+
+        std::wstring str = L"EnableV5Header32bppBGRA";
+        v5HeaderOption.pstrName = (LPOLESTR)str.c_str();
+
+        THROW_IF_FAILED(wicEncoderOptions->Write(1, &v5HeaderOption, &varValue));
+    }
+
+    THROW_IF_FAILED(wicFrameEncoder->Initialize(wicEncoderOptions.get()));
+
+    UINT uWidth, uHeight;
+    THROW_IF_FAILED(wicBitmapSource->GetSize(&uWidth, &uHeight));
+
+    THROW_IF_FAILED(wicFrameEncoder->SetSize(uWidth, uHeight));
+
+    winrt::com_ptr<IWICBitmapSource> wicbitmapSourceConverted;
+    wicbitmapSourceConverted = ConvertWICBitmapPixelFormat(wicImagingFactory, wicBitmapSource, GUID_WICPixelFormat32bppBGRA, WICBitmapDitherTypeNone);
+
+    WICRect rect { 0, 0, static_cast<INT>(uWidth), static_cast<INT>(uHeight) };
+
+    // Write the image data and commit
+    THROW_IF_FAILED(wicFrameEncoder->WriteSource(wicbitmapSourceConverted.get(), &rect));
+
+    THROW_IF_FAILED(wicFrameEncoder->Commit());
+}
+
+winrt::com_ptr<IStream> GetStreamOfWICBitmapSource(
+    winrt::com_ptr<IWICImagingFactory> const& wicImagingFactory,
+    winrt::com_ptr<IWICBitmapSource> const& wicBitmapSource,
+    _In_ REFGUID guidContainerFormat,
+    _In_ BITMAP_VERSION bmpv)
+{
+    winrt::com_ptr<IStream> spImageStream;
+    THROW_IF_FAILED(CreateStreamOnHGlobal(nullptr /* handle */, true /* delete on release */, spImageStream.put()));
+
+    // Create encoder and initialize it
+    winrt::com_ptr<IWICBitmapEncoder> wicEncoder;
+    THROW_IF_FAILED(wicImagingFactory->CreateEncoder(guidContainerFormat, nullptr, wicEncoder.put()));
+
+    THROW_IF_FAILED(wicEncoder->Initialize(spImageStream.get(), WICBitmapEncoderCacheOption::WICBitmapEncoderNoCache));
+
+    // Add a single frame to the encoder with the Bitmap
+    AddFrameToWICBitmap(wicImagingFactory, wicEncoder, wicBitmapSource, bmpv);
+
+    THROW_IF_FAILED(wicEncoder->Commit());
+
+    // Seek the stream to the beginning and transfer
+    static const LARGE_INTEGER lnBeginning = {};
+    THROW_IF_FAILED(spImageStream->Seek(lnBeginning, STREAM_SEEK_SET, nullptr /* new seek pointer */));
+
+    return spImageStream;
+}
+
+void SaveImageWithWIC(
+    winrt::com_ptr<IWICImagingFactory> const& wicImagingFactory,
+    winrt::com_ptr<IWICBitmapSource> const& wicBitmapSource,
+    _In_ REFGUID guidContainerFormat,
+    winrt::com_ptr<IStream>& pStream)
+{
+    winrt::com_ptr<IStream> spImageStream;
+    spImageStream = GetStreamOfWICBitmapSource(wicImagingFactory, wicBitmapSource, guidContainerFormat, BMPV_1);
+
+    // Seek the stream to the beginning and transfer
+    static LARGE_INTEGER const lnBeginning{ 0 };
```

Comment:
Can you explain why this is static? Is it because we are calling this in some loop somehow and need to maintain state?